ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -362,10 +336,14 @@ export class AccountCreation extends React.Component<AccountCreationProps, Accou
   }
 
   clearInstitutionAffiliation() {
-    this.updateInstitutionAffiliation('nonAcademicAffiliation', '');
-    this.updateInstitutionAffiliation('role', '');
-    this.updateInstitutionAffiliation('institution', '');
-    this.updateInstitutionAffiliation('other', '');
+    this.setState(currentState => {
```

Comment:
nit: In general I think it is best practice to avoid directly mutating the innards of the state object, though this is probably a pretty innocuous case since it's happening in the state callback. I think the preferred approach here would be either lodash `set` or via shallow copy:
```
return {
   ...currentState.profile.institutionalAffiliations[0],
  role: '',
  institution: '',
  other: ''
}
```

Reply:
My typical flow is:
- Read normal lodash docs
- Google lodash fp playground and poke at it until I can confirm understanding

Sometimes the mapping from lodash to a side-effect free functional interface is straightforward, other times it's very non-obvious.

Also there are some docs here, but I don't remember the quality:
https://gist.github.com/jfmengels/6b973b69c491375117dc